gateway/src/resty/http/proxy.lua

```diff
@@ -133,15 +31,94 @@ end
 local function connect(request, skip_https_connect)
     local httpc = http.new()
     local proxy_uri = find_proxy_url(request)
+    local uri = request.uri
+    local scheme = uri.scheme
+    local host = uri.host
+    local port = default_port(uri)
 
-    request.ssl_verify = request.options and request.options.ssl and request.options.ssl.verify
-    request.proxy = proxy_uri
+    -- set ssl_verify: lua-resty-http set ssl_verify to true by default if scheme is https, whereas
+    -- openresty treat nil as false, so we need to explicitly set ssl_verify to false if nil
+    local ssl_verify = request.options and request.options.ssl and request.options.ssl.verify or false
+
+    local options = {
+      scheme = scheme,
+      host = host,
+      port = port
+    }
+    if scheme == 'https' then
+        options.ssl_server_name = host
+        options.ssl_verify = ssl_verify
+    end
 
+    -- Connect via proxy
     if proxy_uri then
-        return connect_proxy(httpc, request, skip_https_connect)
+        if proxy_uri.scheme ~= 'http' then
+            return nil, 'proxy connection supports only http'
+        else
+            proxy_uri.port = default_port(proxy_uri)
+        end
+
+        -- Resolve the proxy IP/Port
+        local proxy_host, proxy_port = httpc:resolve(proxy_uri.host, proxy_uri.port)
+        local proxy_url = format("%s://%s:%s", proxy_uri.scheme, proxy_host, proxy_port)
+        local proxy_auth = request.proxy_auth
+
+        if scheme == 'http' then
```

Comment:
`scheme` will always be `https` right?

Reply:
I also think it is always `https` but this is also used by `http_ng` module so let me run a few tests to check if we can remove `http` from this block

Reply:
so this `http` block is never used. I will vote to remove it, what do you think?

Reply:
LGTM

If the scheme is `http`, the code will return with an error

Reply:
The `http` scheme is handled [here](https://github.com/3scale/APIcast/blob/master/gateway/src/apicast/http_proxy.lua#L213), perhaps I missed something.

Would you mind sharing the test steps that led to the error?

Reply:
That's my point, when upstream is plain http, custom http client is not being used. Then, that branch can be removed.

My comment above is that if for some reason (future change) a request to http reaches, the `else` branch will return an error.

Reply:
It turns out that the `http` block is need when using with `http_ng` backend to send request through proxy